arangod/GeneralServer/HttpCommTask.cpp

```diff
@@ -608,12 +619,16 @@ void HttpCommTask::processRequest(std::unique_ptr<HttpRequest> request) {
         << "\"http-request-begin\",\"" << (void*)this << "\",\""
         << _connectionInfo.clientAddress << "\",\""
         << HttpRequest::translateMethod(_requestType) << "\",\""
-        << HttpRequest::translateVersion(_protocolVersion) << "\",\"" << _fullUrl
+        << HttpRequest::translateVersion(_protocolVersion) << "\",\""
+        << (Logger::logFullUrl() 
+             ? _fullUrl 
+             : _fullUrl.substr(0, _fullUrl.find_first_of('?')))
         << "\"";
 
     std::string const& body = request->body();
 
-    if (!body.empty() && Logger::isEnabled(LogLevel::TRACE, Logger::REQUESTS)) {
+    if (!body.empty() && Logger::isEnabled(LogLevel::TRACE, Logger::REQUESTS)
+          && Logger::logFullUrl()) {
```

Comment:
It's not really intuitive here that the option also controls logging the request body.
I think the special behavior here mandates a code comment. Or renaming the option to `--log.request-details` or something more appropriate.